src/main/java/se/lth/cs/connect/routes/Entry.java

```diff
@@ -372,15 +371,7 @@ protected void setup(PippoSettings conf) {
             JcNode unode = new JcNode("user");
             JcNode coll = new JcNode("collection");
 
-            if (collectionId == -1 && newCollectionName != null) {
-                JcNumber cid = new JcNumber("id");
-                collectionId = ((java.math.BigDecimal)(Database.query(rc.getLocal("db"), new IClause[] {
-                    MATCH.node(unode).label("user").property("email").value(user.email),
-                    CREATE.node(unode).relation().out().type("MEMBER_OF")
-                        .node(coll).label("collection").property("name").value(newCollectionName),
-                    RETURN.value(coll.id()).AS(cid)
-                }).resultOf(cid).get(0))).intValue();
-            } else {
+            if(collectionId != -1){
```

Comment:
I think we can remove the if statement here. They way I see it we cover all cases:

 - `e.collection` is a number
  - That is -1 --> `collectionId = user.defaultCollection`
  - Other --> `collectionId = Integer.parseInt(e.collection)`
- `e.collection` is a string
 - `throw new RequestException(..)`


We could also change the condition to `collectionId != user.defaultCollection` as the user probably has access to the default collection.